src/components/charts/Vis7.js

```diff
@@ -81,27 +80,37 @@ export const options = {
   },
 }
 
+const URL1 = 'http://localhost:3000/api/visualization6'
+const URL2 = 'http://localhost:3000/api/visualization7'
 function Vis7() {
   const [data, setData] = useState({ tempData: [], co2Data: [] })
   const [datasets, setDatasets] = useState([])
 
   useEffect(() => {
-    let newCo2Data = CO2DATA.map((item) => {
-      let newItem = {}
-      newItem['x'] = item['Age, gas, calendar years before present (y)']
-      newItem['y'] = item['CO2 concentration (ppm)']
-      return newItem
-    })
-    let newTempData = TEMPDATA.map((item) => {
-      let newItem = {}
-      newItem['x'] = item['Time (kyr BP)'] * 1000
-      newItem['y'] = item['50%']
-      return newItem
-    })
-    setData({ tempData: newTempData, co2Data: newCo2Data })
+    const req1 = axios.get(URL1)
+    const req2 = axios.get(URL2)
+    axios
+      .all([req1, req2])
+      .then(
+        axios.spread((...response) => {
+          let data1 = response[0].data.data
+          let data2 = response[1].data.data
+          setData({ tempData: data1, co2Data: data2 })
+          generateAnnualData()
+        })
+      )
+      .catch((e) => {
+        alert(e)
+      })
+  }, [])
+
+  useEffect(() => {
     generateAnnualData()
   }, [data])
```

Comment:
This useffect will be always executing -> slow
Why? You set data inside it and it is automatically executed, when data changed so you set data, call useeffect, set data, cal useffect, setdata. . .